services/rfq/relayer/inventory/rebalance.go

```diff
@@ -27,218 +27,210 @@ type RebalanceManager interface {
+	maintenancePctDest, err := cfg.GetMaintenanceBalancePct(rebalance.DestMetadata.ChainID, rebalance.DestMetadata.Addr.Hex())
 	if err != nil {
 		return nil, fmt.Errorf("could not get maintenance pct: %w", err)
 	}
-	initialPct, err := cfg.GetInitialBalancePct(destTokenData.ChainID, destTokenData.Addr.Hex())
+	initialPctDest, err := cfg.GetInitialBalancePct(rebalance.DestMetadata.ChainID, rebalance.DestMetadata.Addr.Hex())
 	if err != nil {
 		return nil, fmt.Errorf("could not get initial pct: %w", err)
 	}
 
 	// calculate maintenance threshold relative to total balance
-	tokenName := originTokenData.Name
-	totalBalance := big.NewInt(0)
-	for _, tokenMap := range tokens {
-		for _, tokenData := range tokenMap {
-			if tokenData.Name == tokenName {
-				totalBalance.Add(totalBalance, tokenData.Balance)
-			}
-		}
-	}
-	maintenanceThresh, _ := new(big.Float).Mul(new(big.Float).SetInt(totalBalance), big.NewFloat(maintenancePct/100)).Int(nil)
+	totalBalance := getTotalBalance(cfg, tokens, rebalance.OriginMetadata.Name, rebalance.Method)
+	maintenanceThreshDest, _ := new(big.Float).Mul(new(big.Float).SetInt(totalBalance), big.NewFloat(maintenancePctDest/100)).Int(nil)
 	if span != nil {
-		span.SetAttributes(attribute.Float64("maintenance_pct", maintenancePct))
-		span.SetAttributes(attribute.Float64("initial_pct", initialPct))
-		span.SetAttributes(attribute.String("max_token_balance", originTokenData.Balance.String()))
-		span.SetAttributes(attribute.String("min_token_balance", destTokenData.Balance.String()))
+		span.SetAttributes(attribute.Float64("maintenance_pct_dest", maintenancePctDest))
+		span.SetAttributes(attribute.Float64("initial_pct_dest", initialPctDest))
+		span.SetAttributes(attribute.String("max_token_balance", rebalance.OriginMetadata.Balance.String()))
+		span.SetAttributes(attribute.String("min_token_balance", rebalance.DestMetadata.Balance.String()))
 		span.SetAttributes(attribute.String("total_balance", totalBalance.String()))
-		span.SetAttributes(attribute.String("maintenance_thresh", maintenanceThresh.String()))
+		span.SetAttributes(attribute.String("maintenance_thresh_dest", maintenanceThreshDest.String()))
 	}
 
 	// no need to rebalance if we are not below maintenance threshold on destination
-	if destTokenData.Balance.Cmp(maintenanceThresh) > 0 {
+	if rebalance.DestMetadata.Balance.Cmp(maintenanceThreshDest) > 0 {
 		return nil, nil
 	}
 
-	// calculate the amount to rebalance vs the initial threshold on destination
-	initialThresh, _ := new(big.Float).Mul(new(big.Float).SetInt(totalBalance), big.NewFloat(initialPct/100)).Int(nil)
-	amount = new(big.Int).Sub(originTokenData.Balance, initialThresh)
-
-	// no need to rebalance since amount would not be positive
+	// initially, set the rebalance amount such that it would take origin to the initial threshold
+	initialThreshOrigin, _ := new(big.Float).Mul(new(big.Float).SetInt(totalBalance), big.NewFloat(initialPctDest/100)).Int(nil)
```

Comment:
@dwasse based on https://github.com/synapsecns/sanguine/pull/3003#discussion_r1725744639 we should multiply by `initialPctOrigin` rather than `initialPctDest` here.